graylog2-web-interface/src/views/components/searchbar/completions/FieldValueCompletion.ts

```diff
@@ -0,0 +1,159 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import { isEmpty } from 'lodash';
+import * as Immutable from 'immutable';
+
+import fetch from 'logic/rest/FetchProvider';
+import { qualifyUrl } from 'util/URLUtils';
+import type { TimeRange, NoTimeRangeOverride } from 'views/logic/queries/Query';
+import { ViewMetadataStore } from 'views/stores/ViewMetadataStore';
+import { FieldTypesStore, FieldTypesStoreState, FieldTypeMappingsList } from 'views/stores/FieldTypesStore';
+import FieldTypeMapping from 'views/logic/fieldtypes/FieldTypeMapping';
+
+import type { Completer } from '../SearchBarAutocompletions';
+import type { Token } from '../ace-types';
+
+type SuggestionsResponse = {
+  field: string,
+  input: string,
+  suggestions: Array<{ value: string, occurrence: number}> | undefined,
+}
+
+const suggestionsUrl = qualifyUrl('/search/suggest');
+
+const formatValue = (value: string, type: string) => {
+  if (type === 'constant.numeric') {
+    return Number(value);
+  }
+
+  return value;
+};
+
+const getFieldNameAndInput = (currentToken: Token | undefined | null, lastToken: Token | undefined | null) => {
+  if (currentToken?.type === 'keyword' && currentToken?.value.endsWith(':')) {
+    return {
+      fieldName: currentToken.value.slice(0, -1),
+      input: '',
+    };
+  }
+
+  if (currentToken?.type === 'term' && lastToken?.type === 'keyword') {
+    return {
+      fieldName: lastToken.value.slice(0, -1),
+      input: formatValue(currentToken.value, currentToken.type),
+    };
+  }
+
+  return {};
+};
+
+const getFieldByName = (fields: FieldTypeMappingsList, fieldName: string) => {
+  return fields.find(({ name }) => name === fieldName);
+};
+
+const isEnumerableField = (field: FieldTypeMapping) => {
+  return !!field?.type.properties.find((property) => property === 'enumerable');
+};
+
+class FieldValueCompletion implements Completer {
+  activeQuery: string;
+
+  allFields: FieldTypeMappingsList;
+
+  currentQueryFields: FieldTypeMappingsList;
+
+  fields: FieldTypesStoreState;
+
+  constructor() {
+    this.onViewMetadataStoreUpdate(ViewMetadataStore.getInitialState());
+    ViewMetadataStore.listen(this.onViewMetadataStoreUpdate);
+
+    this._newFields(FieldTypesStore.getInitialState());
+    FieldTypesStore.listen((newState) => this._newFields(newState));
+  }
+
+  shouldShowSuggestions = (fieldName: string) => {
+    if (!fieldName) {
+      return false;
+    }
+
+    const queryField = getFieldByName(this.currentQueryFields, fieldName);
+
+    if (!queryField || !isEnumerableField(queryField)) {
+      const allFieldType = getFieldByName(this.allFields, fieldName);
+
+      return isEnumerableField(allFieldType);
+    }
```

Comment:
I am not completely sure about the best way to identify the most accurate field type. The problem:

The query field types are only being updated, once we submit e.g. the search bar form. When a user changes the streams filter and adjusts the query afterwards (which could trigger the field value suggestions) without submitting the form in the meantime, the query fields are not "up to date". Because of this circumstance I am checking the "all" field types as a fallback, but I can imagine that this can still result in an API error in some cases when the field type properties vary.

Reply:
I think we need to make some improvements for this outside of this PR. Can you create an issue so we can track it?

Reply:
Sounds good, I will create an issue.